### PR TITLE
Pause Substrate sync

### PR DESCRIPTION
**Pull request by nazar-pc**
The first commit was sent upstream here: https://github.com/paritytech/substrate/pull/14465

The second commit makes `sync_mode` updatable and third introduces `SyncMode::Paused` that temporarily makes node not request blocks from other nodes, we'll use that to pause regular Substrate sync when doing sync from DSN (otherwise it results in a decreased reputation of connected peers and a lot of unnecessary reconnections/disconnections).

I'll try to upstream the other two commits if/when the first one lands upstream.

I have tested this locally with Subspace node, seems to behave as expected.

Needed for https://github.com/subspace/subspace/issues/1514

### CODE REVIEW

File: client/network/sync/src/engine.rs [hunk 1]
```diff
@@ -498,19 +487,6 @@ where
 						self.block_announce_data_cache.put(announce.header.hash(), data);
 					}
 				}
-
-				return
-			},
-			PollBlockAnnounceValidation::ImportHeader { announce, is_best, who } => {
```
Comment (shamil-gadelshin) [inline client/network/sync/src/engine.rs:504]: I wonder, why did you remove this option?

**Review by shamil-gadelshin — APPROVED** [review]
The PR generally makes sense to me.

File: client/network/sync/src/engine.rs (hunk 1 above)
Comment (nazar-pc) [reply client/network/sync/src/engine.rs:504]: It is a part of https://github.com/paritytech/substrate/pull/14465 and it was no longer used after `SyncMode::Light` enum variant was gone, I just pulled the strings and removed all unused code after that.
